src/tools/compiletest/src/runtest.rs

```diff
@@ -2749,7 +2749,7 @@ impl<'test> TestCx<'test> {
         // Wrapper tools set by `runner` might provide extra output on failure,
         // for example a WebAssembly runtime might print the stack trace of an
         // `unreachable` instruction by default.
-        let compare_output_by_lines = self.config.runner.is_some();
+        let compare_output_by_lines = self.props.parallel_front_end || self.config.runner.is_some();
```

Comment:
If this is the only thing that `parallel_front_end` enables, maybe exposing `//@ compare-output-by-lines` would be better?